sys/net/iflib.c

```diff
@@ -960,6 +960,7 @@ iflib_netmap_txsync(struct netmap_kring *kring, int flags)
 	u_int const lim = kring->nkr_num_slots - 1;
 	u_int const head = kring->rhead;
 	struct if_pkt_info pi;
+	int pkt_sent, bytes_sent;
```

Comment:
just for consistency can we make these tx_pkts and tx_bytes?

Reply:
> Oh, nice find. Is this just an issue the way we set up IPS mode (from host to tx and from rx to host) or does the problem also exist when "bridging" interfaces?

No, IPS is correctly being setup, that is, setup as intended by the author of Netmap (e.g. igb0 -> igb0^). The actual problem resides in the TX-RX datapath. As such, you can expect this problem to persist when setting up a netmap bridge. Basically the system searches for the drivers' internal function that provides hardware statistics, if certain statistics are not present, it defaults to a default counter that is ONLY being tracked when a packet goes through the non-netmap flow. I simply added the counter in the netmap-specific TX-RX functions :-) 

The code I'm providing here prevents a driver author from having to implement counting mandatory statistics which might not even be available on the hardware.

In fact, regarding the two options I described earlier, using option 1 will cause a bridged interface to NOT report packets and bytes, while option 2 will.

> just for consistency can we make these tx_pkts and tx_bytes?

Of course! I just copied the variable names used for the counters in the non-netmap TX-RX